data_steward/cdr_cleaner/cleaning_rules/valid_death_dates.py

```diff
@@ -1,53 +1,142 @@
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = 'All rows in the death table that contain a death_date which occurs before the start of the AoU program' \
+               '(Jan 1, 2017) or after the current date will be sandboxed and dropped'
+        super().__init__(issue_numbers=['DC822'],
+                         description=desc,
+                         affected_datasets=[cdr_consts.COMBINED],
+                         affected_tables=death,
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id)
+
+    def get_query_specs(self, *args, **keyword_args):
+        """
+        Return a list of dictionary query specifications.
+
+        :return:  A list of dictionaries. Each dictionary contains a single query
+            and a specification for how to execute that query. The specifications
+            are optional but the query is required.
+        """
+        keep_valid_death_dates = {
+            cdr_consts.QUERY:
+                KEEP_VALID_DEATH_DATE_ROWS.render(
+                    project_id=self.project_id,
+                    dataset_id=self.dataset_id,
+                    table=death,
+                    sandbox_id=self.sandbox_dataset_id,
+                    sandbox_table=self.get_sandbox_table_for(death)),
+            cdr_consts.DESTINATION_TABLE:
+                death,
+            cdr_consts.DESTINATION_DATASET:
+                self.dataset_id,
+            cdr_consts.DISPOSITION:
+                WRITE_TRUNCATE
+        }
+
+        sandbox_invalid_death_dates = {
+            cdr_consts.QUERY:
+                SANDBOX_INVALID_DEATH_DATE_ROWS.render(
+                    project_id=self.project_id,
+                    sandbox_id=self.sandbox_dataset_id,
+                    sandbox_table=self.get_sandbox_table_for(death),
+                    dataset_id=self.dataset_id,
+                    table=death,
+                    program_start_date=program_start_date,
+                    current_date=current_date)
+        }
+
+        return [keep_valid_death_dates, sandbox_invalid_death_dates]
+
+    def setup_rule(self, client, *args, **keyword_args):
+        """
+        Function to run any data upload options before executing a query.
+        """
+        pass
+
+    def get_sandbox_tablenames(self):
+        """
+        Get the sandbox dataset id for this class instance
+        """
+        return [
+            self.get_sandbox_table_for(affected_table)
+            for affected_table in self.affected_tables
+        ]
+
+    def get_sandbox_table_for(self, affected_table):
```

Comment:
This is a redundant method, you could use the same logic in the get_sandbox_tablenames() method instead of here. get_sandbox_tablenames() will be refactored as part of using shared sandbox datasets epic. it will be easier then, to update get_sandbox_tablenames() in all the classes.

Reply:
Ah yes, this was pushed before we talked about this and I forgot to make those changes afterwards 😅 I'll modify it and repush 👍🏾